blog/_posts/2017-09-04-gsoc-NeuralNetDiffEq.md

```diff
@@ -0,0 +1,172 @@
+
+## How to use Neural Network for solving Differential Equations?
+The concept of this solver is based on the UAT (Universal Approximation Theorem) which says that a Neural Network with not more than 3 hidden can approximate any continuous function. The Neural Network is made to minimize a loss function which then results in the convergence of our Trial solution towards the actual (analytical) solution. To know more about UAT [click here](http://neuralnetworksanddeeplearning.com/chap4.html).
+
+## Research aspect of the project and The Challenge
+The research paper we referred on the topic is quite old and lacks proper explanation as well as examples. The research done on this part of Neural Networks is not too much and we were not able to get much help from the research papers related to the topic. The initial task was to read and understand the mathematics behind solving Differential Equations. Also the (analytical) methods used to solve Differential Equations on computers are quite different from the ones we use on paper so it took quite some time to get familiar with them. Also the structure and type of NN to be used so that the solver advantages are retained without compromising the performance was a research subdomain as well as a challenge.
+
+Also after implementing the solver for ODEs (Ordinary Differential Equations) and System of ODEs the difficult part was to make the Neural Network converge, especially for the system of ODEs. As there are a lot of factors involved in Neural Networks like Hidden Layer Width, Number of Hidden Neurons, Activations, Weights etc., I relied on my Machine Learning background as well as the help from my mentors to experiment with most of the feasible settings of NN hyper-parameters and recording the accuracy of convergence and performance of the solver.
+
+Making NN converge for System of ODEs is not as easy as it seems and took up most of the time for experimentation and tuning. Predicting the system of DEs solution with larger domain is still a challenge which needs to be worked upon.
```

Comment:
"Predicting the system of DEs solution with larger domain is still a challenge which needs to be worked upon.". I don't understand what you mean here.

Reply:
I was referring to the ode problem domain which we solve for
Example:
`prob_ode_lotkavoltera = ODEProblem(lotka_volterra,Float32[1.0,1.0],(Float32(0.0),Float32(1.0)))`
and 
`prob_ode_lotkavoltera = ODEProblem(lotka_volterra,Float32[1.0,1.0],(Float32(0.0),Float32(10.0)))
`

Reply:
You might want to fully highlight that with that example.
Eg showing "While lotka-voltera works over the short domain of 0-1 (Plot), we can see it fails over a larger domain e.g. 0-10 (Plot)"

Nothing is wrong with highlighting the failure of the method clearly.

Reply:
Alright!